miden-tx/src/host/mod.rs

```diff
@@ -483,52 +484,64 @@ impl<A: AdviceProvider> Host for TransactionHost<A> {
     }
 
     fn on_event(&mut self, process: ProcessState, event_id: u32) -> Result<(), ExecutionError> {
-        let event = TransactionEvent::try_from(event_id)
-            .map_err(|err| ExecutionError::EventError(Box::new(err)))?;
-
-        if process.ctx() != ContextId::root() {
-            return Err(ExecutionError::EventError(Box::new(
-                TransactionEventError::NotRootContext(event_id),
-            )));
-        }
+        // if the prefix of the event ID is not equal to 2, try to parse it as a library event
+        if event_id >> 16 != EVENT_ID_PREFIX {
+            let system_event =
+                SystemEvent::from_event_id(event_id).ok_or(ExecutionError::EventError(
+                    Box::new(TransactionEventError::InvalidTransactionEvent(event_id)),
+                ))?;
+            if system_event == SystemEvent::FalconSigToStack {
+                self.on_signature_requested(process)?
+            }
```

Comment:
If we parsed a system event, but it wasn't `FalconSigToStack`, we should probably return an error.

Overall, we could probably change this to something like:
```rust
if Ok(system_event) = SystemEvent::from_event_id(event_id) {
    if system_event == SystemEvent::FalconSigToStack {
        return self.on_signature_requested(process);
    } else {
        // TODO: return error
    }
}
```
And then the rest of the code would remain the same.